src/hooks/useExportSatelliteManifest.ts

```diff
@@ -0,0 +1,99 @@
+import { useQuery, QueryObserverResult } from 'react-query';
+import Cookies from 'js-cookie';
+import { getConfig } from '../utilities/platformServices';
+
+interface TriggerManifestExportResponse {
+  body: {
+    exportJobID: string;
+    href: string;
+  };
+}
+
+interface ExportManifestStatusResponse {
+  body: {
+    exportID: string;
+    href: string;
+  };
+}
+
+const triggerManifestExport = (uuid: string): Promise<TriggerManifestExportResponse> => {
+  const cs_jwt = Cookies.get('cs_jwt');
+  const { rhsmAPIBase } = getConfig();
+  return fetch(`${rhsmAPIBase}/management/v1/allocations/${uuid}/export`, {
+    headers: { Authorization: `Bearer ${cs_jwt}` },
+    mode: 'cors'
+  }).then((response) => {
+    return response.json();
+  });
+};
+
+const sleep = (milliseconds: number) => {
+  return new Promise((resolve) => setTimeout(resolve, milliseconds));
+};
+
+const getManifestExportStatus = async (
+  uuid: string,
+  exportJobID: string
+): Promise<ExportManifestStatusResponse> => {
+  const cs_jwt = Cookies.get('cs_jwt');
+  const { rhsmAPIBase } = getConfig();
+  const response = await fetch(
+    `${rhsmAPIBase}/management/v1/allocations/${uuid}/exportJob/${exportJobID}`,
+    {
+      headers: { Authorization: `Bearer ${cs_jwt}` },
+      mode: 'cors'
+    }
+  );
+
+  if (response.status === 200) {
+    return response.json();
+  }
+
+  if (response.status === 202 || response.status === 404) {
+    // Export not ready, we need to pause and retry.
+    await sleep(5000);
+    return getManifestExportStatus(uuid, exportJobID);
+  }
+
+  throw new Error('Error fetching status of exported manifest');
+};
+
+const downloadExportedManifest = (uuid: string, exportID: string): Promise<Blob> => {
+  const cs_jwt = Cookies.get('cs_jwt');
+  const { rhsmAPIBase } = getConfig();
+  return fetch(`${rhsmAPIBase}/management/v1/allocations/${uuid}/export/${exportID}`, {
+    headers: { Authorization: `Bearer ${cs_jwt}`, 'Content-type': 'application/zip' },
+    mode: 'cors'
+  })
+    .then((response) => response.blob())
+    .then((zipFile) => zipFile);
+};
+
+const exportManifest = async (uuid: string): Promise<Blob> => {
+  const triggerResponse = await triggerManifestExport(uuid);
+
+  const statusResponse = await getManifestExportStatus(uuid, triggerResponse.body.exportJobID);
+
+  const downloadResponse = await downloadExportedManifest(uuid, statusResponse.body.exportID);
+
+  return downloadResponse;
+};
+
+const useExportSatelliteManifest = (
+  uuid: string,
+  shouldLoadOnRender: boolean
+): QueryObserverResult<Blob, Error> => {
+  return useQuery<Blob, Error>(['exportedManifests', uuid], () => exportManifest(uuid), {
+    enabled: shouldLoadOnRender,
+    retryDelay: 5 * 60 * 1000
```

Comment:
Ah yeah so I wasn't 100% on what the max amount of time it takes to export a manifest -- in John's account, for ex, there is a manifest called aspurrie_sil, and it takes about 40 seconds for the export to be ready.

I went ahead and changed it to 1 minute. Not sure if anyone on the team might know if it should be longer for some users or not

Reply:
sooo...there are some customers that have manifests that can take 10 minutes to generate download.  I don't have any accounts on hand, but we had to update timeouts to the maximum allowable so that these customers could download their manifests in RHSM Web.

RHSM Web uses just the one synchronus manifest endpoint (which is deprecated in favor of doing async generation/download).

Reply:
aha ok so maybe I should set the retry delay to 10 minutes then, just to be safe?

Reply:
Per discussion, I updated this so that retries are off for this query.